lib/charms/opensearch/v0/opensearch_users.py

```diff
@@ -251,17 +254,33 @@ def remove_users_and_roles(self, departed_relation_id: Optional[int] = None):
         """
         if not self.opensearch.is_node_up() or not self.unit.is_leader():
             return
+        rel_id = str(departed_relation_id)
 
-        relations = self.model.relations.get(ClientRelationName, [])
-        relation_users = set(
-            [
-                f"{ClientRelationName}_{relation.id}"
-                for relation in relations
-                if relation.id != departed_relation_id
-            ]
-        )
-        self._remove_lingering_users(relation_users)
-        self._remove_lingering_roles(relation_users)
```

Comment:
I believe we can keep it simple and keep this and remove the newly added code, and simply add `rel_id` to the `_remove_lingering_*` methods.

Reply:
Fully re-fatctored (into `opensearch_provider` ). Closing thread as it's outdated, new one may raise relating to actual code.